build_tools/scripts/check_path_lengths.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This scans the IREE source tree for long path lengths, which are problematic
+# on Windows: https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation
+#
+# To check that all relative paths are shorter than the default limit:
+#   python check_path_lengths.py
+#
+# To check that all relative paths are shorter than a custom limit:
+#   python check_path_lengths.py --limit=50
+
+import argparse
+import os
+import sys
+
+
+def repo_relpath(repo_root, path):
+  return os.path.relpath(path, repo_root).replace("\\", "/")
+
+
+def parse_arguments():
+  parser = argparse.ArgumentParser(description="Path length checker")
+  parser.add_argument("--limit",
+                      help="Path length limit (inclusive)",
+                      type=int,
+                      default=75)
+  parser.add_argument("--include_tests",
+                      help="Includes /test directories",
+                      action="store_true",
+                      default=False)
+  args = parser.parse_args()
+  return args
+
+
+def main(args):
+  repo_root = os.path.dirname(
+      os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```

Comment:
I hope one day we'll have a consistent and robust way to do this. I still prefer shelling out to `git rev-parse --show-toplevel`. But if you're going with path manipulation, I think pathlib is much less verbose here.

```suggestion
  repo_root = pathlib.Path(__file__).parent.parent.parent
```